### ISSUE TITLE
Support "early return" version of `if-else` statements

### ISSUE THREAD

**Issue by smores56** [roc-lang/roc#7039]
We want to allow users to have early returns in their functions. Basically, we want to support the current syntax:

```roc
foo = \list ->
    if List.isEmpty list then
        baz 42 blah etc
    else
        when List.len list is
```

as well as this:

```roc
foo = \list ->
    if List.isEmpty list then
        baz 42 blah etc
        else

    when List.len list is
```

We want to force the early return expression onto its own, indented block (a.k.a. it can't be right after the `then`), and we want to force the empty line after that block for readability's sake.

_implementation suggestion: store whether the `else` was indented in the AST, if it was not indented use the original formatting, otherwise use the new formatting._

**Comment by lishaduck** [comment]
Closed in #7060?

**Comment by Anton-4** [comment]
Yes, thanks @lishaduck :)
